### PR TITLE
Fix unroll_nested_function_calls in export utils (Fixes #190)

### LINKED ISSUE #190: TPOT.export function is unable to dump the optimised pipeline
This is in relation to TPOT release version 0.4 

TPOT generated output pipeline - has no information on classifier. 
The behavior is seen when feature selection is done with SelectPercentile classifier and another classifier model fitted training set.
The below is a sample TPOT generated pipeline where the classifier information is missing.
 
=====================================================================
import numpy as np
import pandas as pd

from sklearn.cross_validation import train_test_split
from sklearn.feature_selection import SelectPercentile, f_classif

'# NOTE: Make sure that the class is labeled 'class' in the data file
tpot_data = pd.read_csv('PATH/TO/DATA/FILE', sep='COLUMN_SEPARATOR')
training_indices, testing_indices = train_test_split(tpot_data.index, stratify = tpot_data['class'].values, train_size=0.75, test_size=0.25)

result1 = tpot_data.copy()

'#Use Scikit-learn's SelectPercentile for feature selection
training_features = result1.loc[training_indices].drop('class', axis=1)
training_class_vals = result1.loc[training_indices, 'class'].values

if len(training_features.columns.values) == 0:
    result1 = result1.copy()
else:
    selector = SelectPercentile(f_classif, percentile=40)
    selector.fit(training_features.values, training_class_vals)
    mask = selector.get_support(True)
    mask_cols = list(training_features.iloc[:, mask].columns) + ['class']
    result1 = result1[mask_cols]

=====================================================================

On debugging further with adding prints in export_utils.py found the following 

**Best pipeline: _extra_trees(_select_percentile(input_df, 40), 32, 0.62, 0.45000000000000001)**
0.696427835936
node_params
[<deap.gp.Terminal object at 0x7f3372a7a140>, <deap.gp.Terminal object at 0x7f3372a8f320>]
new_val is
result1
operator_list
**['result1', '_select_percentile']**
pipeline_list
**[['result1', '_select_percentile', 'ARG0', '40']]**
node_params
[<deap.gp.Terminal object at 0x7f33717998c0>, <deap.gp.Terminal object at 0x7f3372a8f0a0>, <deap.gp.Terminal object at 0x7f3372a91e10>, <deap.gp.Terminal object at 0x7f3372a918c0>]
new_val is
result2
operator_list
**['result2', '_extra_trees']**
pipeline_list
[['result1', '_select_percentile', 'ARG0', '40'], ['result2', '_extra_trees', 'result1', '32', '0.62', '0.45']]
linesdasdf node_index
0
line40 - node
<deap.gp.Terminal object at 0x7f3371799500>
pipeline_list[0] - ankit
**['result1', '_select_percentile', 'ARG0', '40']**
x is
result1
x is
**_select_percentile**

Carefully observing the bold parts we see for result2 also result1 object is being picked causing extra_trees not being picked and thus the incomplete pipeline dump.

### PR DESCRIPTION
**Pull request by teaearlgraycold**
## What does this PR do?

Fixes a bug in the `unroll_nested_function_calls` function and adds a test to prevent similar issues in the future.

## What are the relevant issues?

#190

### CODE REVIEW

File: tpot/export_utils.py [hunk 1]
```diff
@@ -61,7 +61,8 @@ def unroll_nested_fuction_calls(exported_pipeline):
             break
 
     # Replace 'ARG0' with 'input_df'
-    pipeline_list = [[x if x != 'ARG0' else 'input_df' for x in pipeline_list[0]]]
+    for index in range(len(pipeline_list)):
```
Comment (teaearlgraycold) [inline tpot/export_utils.py]: This is the the actual fix here

File: tpot/export_utils.py [hunk 2]
```diff
@@ -61,7 +61,8 @@ def unroll_nested_fuction_calls(exported_pipeline):
...
+        pipeline_list[index] = [x if x != 'ARG0' else 'input_df' for x in pipeline_list[index]]
```
Comment (teaearlgraycold) [inline tpot/export_utils.py]: This is the the actual fix here

**Comment by ankitshah009** [comment]
The above fix for #190 is not working fine. 
@teaearlgraycold - Is this the only fix done to the code in tpot/export_utils.py? Are there no changes done to the main tpot.py script?

Receiving these errors : - 
pipeline_list
[['result1', '_feat_agg', 'ARG0', '36', '13', '5'], ['result2', '_passive_aggressive', 'result1', '0.08', '11']]
  File "generate_tpot_model.py", line 89, in <module>
    runTpot(train_file, test_file, gen_num)
  File "generate_tpot_model.py", line 80, in runTpot
    tpot.export(name)
  File "/usr/local/lib/python2.7/dist-packages/tpot/tpot.py", line 480, in export
    pipeline_text += replace_function_calls(pipeline_list)
  File "/usr/local/lib/python2.7/dist-packages/tpot/export_utils.py", line 191, in replace_function_calls
    operator_num = int(operator[0].strip('result'))
ValueError: invalid literal for int() with base 10: ' '

**Comment by ankitshah009** [comment]
Another error sample for a different run.
[['result1', '_select_fwe', 'ARG0', '12.0'], ['result2', '_extra_trees', 'result1', '6', '48.0', '0.1']]

Traceback (most recent call last):
  File "generate_tpot_model.py", line 89, in <module>
    runTpot(train_file, test_file, gen_num)
  File "generate_tpot_model.py", line 80, in runTpot
    tpot.export(name)
  File "/usr/local/lib/python2.7/dist-packages/tpot/tpot.py", line 477, in export
    pipeline_text = generate_import_code(pipeline_list)
  File "/usr/local/lib/python2.7/dist-packages/tpot/export_utils.py", line 99, in generate_import_code
    operators_used = set([operator[1] for operator in pipeline_list])
IndexError: string index out of range

**Pull request merged**